oxen-io/session-android
# Fix/phone call from lockscreen

### Description

Fix to prevents failure to display call notifications when the device is asleep or locked. Android 12 and above has tightened permissions with regard to start services, in particularly foreground services in such cases. As such, this patch wakes the device up in order to start the call service and display an incoming call notification.

### Test steps:
With two devices running (Alice and Bob) - have Alice call Bob under the following conditions:
- With Bob having the app visible in the foreground,
- With Bob having the app running the background (i.e., dismissed but not closed),
- With Bob having force-closed the app (i.e., manually closed it from the back-stack) but the device is awake,
- With Bob having force-closed the app and then locked the device.

In each case Bob should receive an incoming call notification. You may also wish to test with adding a lock-screen to Bob's device (pin or pattern etc.), and optionally locking Session (which will require a second unlock identical to that required to lock the device). Again, in each case Bob should receive an incoming call notification which he can answer or decline.

It should be noted that Calls are in Beta and may not always successfully set up a call, so it may be possible to fail to initialise a call entirely, which may not necessarily be caused by the changes in this branch, which on the whole have been observed to improve the availability of call notifications.
